### ISSUE TITLE
Low KITTI 2015 flow accuracy by using pertained weights

### ISSUE THREAD

**Issue by magsail** [ltkong218/FastFlowNet#22]
Hi there,

I tested the accuracy of the model using pretrained weights in `checkpoints/` on KITTI 2015 flow training dataset,  and got lower accuracy than that reported in your paper.

Using `./checkpoints/fastflownet_ft_mix.pth` gives accuracy of 
```
NOC
0.203071 0.203071 0.284329 0.284329 0.216956 0.216956 1.000000 
OCC
0.301341 0.301341 0.306706 0.306706 0.302163 0.302163 1.000000 
```

`./checkpoints/fastflownet_ft_kitti.pth` gives accuracy of 
```
NOC
0.242663 0.242663 0.172001 0.172001 0.230589 0.230589 1.000000 
OCC
0.310403 0.310403 0.192049 0.192049 0.292272 0.292272 1.000000 
```

The paper shows the model could give (8.21%) Fl-all on KITTI 2015 flow training dataset.

So I'm wondering if the weights that can reproduced the accuracies in paper is included in this repo ? Please clarify.

Thanks so much !

**Comment by magsail** [comment]
Posting the code that I'm using to test on KITTI 2015 Flow training set. There is no changes made to the model or the parameters of the model.

```python

import os
import torch
import torch.nn as nn
import torch.nn.functional as F
import cv2
import numpy as np
from fastflownet import FastFlowNet, centralize

script_root = os.path.dirname(__file__)
from glob import glob
ops_so = glob(os.path.join(script_root,'./build/*/*.so'))[0]
torch.ops.load_library(ops_so)

def save_kitti_flow(save_path, flow):
    flow = torch.squeeze(flow, dim=0)
    h = flow.shape[1]
    w = flow.shape[2]
    valid_channel = np.ones((h, w))
    flow_base = flow * 64 + np.power(2, 15)
    flow_u = flow_base[0, :, :]
    flow_v = flow_base[1, :, :]
    flow_final_base = np.stack((valid_channel, flow_v.cpu(), flow_u.cpu()), axis=2).astype(np.uint16)
    cv2.imwrite(save_path, flow_final_base)

if __name__ == '__main__':

    div_flow = 20.0
    div_size = 64

    model = FastFlowNet().eval()
    model.load_state_dict(torch.load('./checkpoints/fastflownet_ft_kitti.pth'))

    IMG_ROOT = "/data/share/kitti2015/data_scene_flow/training/image_2/"
    OUTPUT_BASE = "./kitti_output/data/flow/"

    with torch.no_grad():
        for i in range(200):
            print("Processing Image: " + str(i))
            img_path_0 = IMG_ROOT + f'{i:06d}' + "_10.png"
            img_path_1 = IMG_ROOT + f'{i:06d}' + "_11.png"
            img1 = torch.from_numpy(cv2.imread(img_path_0)).float().permute(2, 0, 1).unsqueeze(0)/255.0
            img2 = torch.from_numpy(cv2.imread(img_path_1)).float().permute(2, 0, 1).unsqueeze(0)/255.0
            img1, img2, _ = centralize(img1, img2)

            height, width = img1.shape[-2:]
            orig_size = (int(height), int(width))

            if height % div_size != 0 or width % div_size != 0:
                input_size = (
                    int(div_size * np.ceil(height / div_size)), 
                    int(div_size * np.ceil(width / div_size))
                )
                img1 = F.interpolate(img1, size=input_size, mode='bilinear', align_corners=False)
                img2 = F.interpolate(img2, size=input_size, mode='bilinear', align_corners=False)
            else:
                input_size = orig_size

            input_t = torch.cat([img1, img2], 1)

            output = model(input_t).data

            flow = div_flow * F.interpolate(output, size=input_size, mode='bilinear', align_corners=False)

            if input_size != orig_size:
                scale_h = orig_size[0] / input_size[0]
                scale_w = orig_size[1] / input_size[1]
                flow = F.interpolate(flow, size=orig_size, mode='bilinear', align_corners=False)
                flow[:, 0, :, :] *= scale_w
                flow[:, 1, :, :] *= scale_h

            flow = flow[0].cpu()

            output_path = OUTPUT_BASE + f'{i:06d}' + "_10.png"
            save_kitti_flow(output_path, flow)

```
